### PR TITLE
Add GenPrism macro

### PR DESCRIPTION
**Pull request by julien-truffaut**

### CODE REVIEW

**Comment by julien-truffaut** [comment]
solving #172

**Comment by NightRa** [comment]
Missing check for an inheritance relation between the classes.
Otherwise, looks good.

File: macro/src/main/scala/monocle/macros/GenPrism.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package monocle.macros
+
+import monocle.Prism
+import monocle.macros.internal.MacrosCompatibility
+
+
+
+
+object GenPrism {
+  def apply[S, A <: S]: Prism[S, A] = macro GenPrismImpl.genPrism_impl[S, A]
```
Comment (julien-truffaut) [inline macro/src/main/scala/monocle/macros/GenPrism.scala]: @NightRa I am doing the check here

Comment (NightRa) [inline macro/src/main/scala/monocle/macros/GenPrism.scala]: Oh... alright :)
You should probably put a comment saying that.

Comment (julien-truffaut) [inline macro/src/main/scala/monocle/macros/GenPrism.scala]: good point

**Pull request merged**